### PR TITLE
trilium: 0.51.2 -> 0.53.2 + update script

### PR DESCRIPTION
**Pull request by FliegendeWurst**
###### Description of changes
New stable release: https://github.com/zadam/trilium/releases/tag/v0.52.2 https://github.com/zadam/trilium/releases/tag/v0.52.3
Security update: https://github.com/zadam/trilium/releases/tag/v0.52.4
New stable release: https://github.com/zadam/trilium/releases/tag/v0.53.2

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`
- [x] Tested basic functionality of all binary files
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by eliandoran — APPROVED** [review]

**Comment by viq** [comment]
What's going on with this PR? It's a security fix, and it's been hanging here for at least 3 weeks now with no action.

File: pkgs/applications/office/trilium/default.nix [hunk 1]
```diff
@@ -20,46 +20,47 @@ let
...
 in {
 
   trilium-desktop = stdenv.mkDerivation rec {
     pname = "trilium-desktop";
     inherit version;
-    inherit meta;
+    meta = with metaCommon; {
+      inherit description homepage license sourceProvenance platforms maintainers;
+      mainProgram = "trilium";
+    };
```
Comment (SuperSandro2000) [inline pkgs/applications/office/trilium/default.nix:39]:
```suggestion
    meta = metaCommon // {
      mainProgram = "trilium";
    };
```

File: pkgs/applications/office/trilium/default.nix [hunk 2]
```diff
@@ -20,46 +20,47 @@ let
...
       autoPatchelfHook
       makeWrapper
       wrapGAppsHook
+      copyDesktopItems
     ];
 
-    buildInputs = atomEnv.packages ++ [ gtk3 libxshmfence ];
+    buildInputs = atomEnv.packages ++ [ libxshmfence ];
+
+    desktopItems = [ desktopItem ];
```
Comment (SuperSandro2000) [inline pkgs/applications/office/trilium/default.nix:52]: Why not inline this?

Comment (FliegendeWurst) [reply pkgs/applications/office/trilium/default.nix]: Changed to inline definition.

File: pkgs/applications/office/trilium/default.nix [hunk 3]
```diff
@@ -1,17 +1,9 @@
...
-  description = "Trilium Notes is a hierarchical note taking application with focus on building large personal knowledge bases";
-  desktopItem = makeDesktopItem {
-    name = "Trilium";
-    exec = "trilium";
-    icon = "trilium";
-    comment = description;
-    desktopName = "Trilium Notes";
-    categories = [ "Office" ];
-  };
+  description = "Hierarchical note taking application with focus on building large personal knowledge bases";
```
Comment (SuperSandro2000) [inline pkgs/applications/office/trilium/default.nix:5]:
```suggestion
```

File: pkgs/applications/office/trilium/default.nix [hunk 4]
```diff
@@ -1,17 +1,9 @@
...
 
-  meta = with lib; {
+  metaCommon = with lib; {
     inherit description;
```
Comment (SuperSandro2000) [inline pkgs/applications/office/trilium/default.nix:7]:
```suggestion
      description = "Hierarchical note taking application with focus on building large personal knowledge bases";
```

**Pull request merged**